app/code/Magento/SearchStorefrontElasticsearch/Model/Config.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\SearchStorefrontElasticsearch\Model;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\SearchStorefrontElasticsearch\Model\Client\ClientOptionsInterface;
+use Magento\SearchStorefrontElasticsearch\Model\Client\ClientResolver;
+use Magento\SearchStorefrontElasticsearch\Model\ConnectionConfigInterface as ConnectionConfig;
+use Magento\SearchStorefrontStub\Framework\Model\EngineResolverInterface;
+
+/**
+ * Elasticsearch config model
+ * Copy Magento\Elasticsearch\Model\Config removed dependency on store module and read config from deployment
+ */
+class Config implements ClientOptionsInterface
+{
+    /**
+     * Search engine name
+     */
+    const ENGINE_NAME = 'elasticsearch';
+
+    /**
+     * Elasticsearch Entity type
+     */
+    const ELASTICSEARCH_TYPE_DOCUMENT = 'document';
+
+    /**
+     * Elasticsearch default Entity type
+     */
+    const ELASTICSEARCH_TYPE_DEFAULT = 'product';
+
+    /**
+     * Default Elasticsearch server timeout
+     */
+    const ELASTICSEARCH_DEFAULT_TIMEOUT = 15;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
+
+    /**
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @var ClientResolver
+     */
+    private $clientResolver;
+
+    /**
+     * @var EngineResolverInterface
+     */
+    private $engineResolver;
+
+    /**
+     * Available Elasticsearch engines.
+     *
+     * @var array
+     */
+    private $engineList;
+
+    /**
+     * @var ConnectionConfig
+     */
+    private $config;
+
+    /**
+     * @param ScopeConfigInterface    $scopeConfig
+     * @param ClientResolver          $clientResolver
+     * @param EngineResolverInterface $engineResolver
+     * @param ConnectionConfig        $config
+     * @param string|null             $prefix
+     * @param array                   $engineList
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        ClientResolver $clientResolver,
+        EngineResolverInterface $engineResolver,
+        ConnectionConfig $config,
+        $prefix = null,
+        $engineList = []
+    ) {
+        $this->scopeConfig = $scopeConfig;
+        $this->clientResolver = $clientResolver;
+        $this->engineResolver = $engineResolver;
```

Comment:
Property is used only in constructor, perhaps we are dealing with dead code here. 
Please, check other files as well and remove any not used code (Php Inspections 
 or similar plugin can help with it, see https://github.com/kalessil/phpinspectionsea )

        $this->clientResolver = $clientResolver;
        $this->engineResolver = $engineResolver;
        $this->prefix = $prefix ?: $this->clientResolver->getCurrentEngine();
        $this->engineList = $engineList;

Reply:
removed unused code